src/Microsoft.AspNetCore.SignalR/HubEndPoint.cs

```diff
@@ -138,15 +139,26 @@ public class HubEndPoint<THub, TClient> : StreamingEndPoint, IInvocationBinder w
             }
         }
 
-        private async Task DispatchMessagesAsync(StreamingConnection connection)
+        private async Task DispatchMessagesAsync(Connection connection)
         {
-            var stream = connection.Transport.GetStream();
             var invocationAdapter = _registry.GetInvocationAdapter(connection.Metadata.Get<string>("formatType"));
 
-            while (true)
+            while (await connection.Transport.Input.WaitToReadAsync())
             {
-                // TODO: Handle receiving InvocationResultDescriptor
-                var invocationDescriptor = await invocationAdapter.ReadMessageAsync(stream, this) as InvocationDescriptor;
+                Message message;
+                if (!connection.Transport.Input.TryRead(out message))
+                {
+                    continue;
```

Comment:
It can return false due to a race between readers. We're a single-consumer system so it's not really a problem, but that's the theoretical case where it can be false even when WaitToReadAsync unblocks